### PR TITLE
Adds q8

### PR DESCRIPTION
**Pull request by kristina-altunina**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -167,3 +167,46 @@ describe('GET/api/articles/:article_id/comments', () => {
...
+        .then((response) => {
+        const {article} = response.body
+            expect(article).toHaveProperty("title");
+            expect(article).toHaveProperty("topic");
+            expect(article).toHaveProperty("body");
+            expect(article).toHaveProperty("created_at");
+            expect(article).toHaveProperty("votes");
+            expect(article).toHaveProperty("article_img_url");
+        })
+    });
```
Comment (Hard-Luck) [inline __tests__/api.test.js]: When testing the properties we should also test the value of the properties we know, ie article_id and votes 
if you look at jest documentation you will see that toHaveProperty takes a second argument which will help you check the values

File: __tests__/api.test.js [hunk 2]
```diff
@@ -167,3 +167,46 @@ describe('GET/api/articles/:article_id/comments', () => {
...
+        return request(app).patch('/api/articles/1000').send(newVote)
+        .expect(404)
+        .then((result) => {
+            const {body} = result
+            expect(result.status).toBe(404)
+            expect(body.message).toBe('404: Not Found')
+        })
+    });
+});
+
```
Comment (Hard-Luck) [inline __tests__/api.test.js:341]: You should add a test for patch /api/articles/not-a-number

File: controllers/articles-controller.js [hunk 3]
```diff
@@ -23,4 +23,21 @@ const getArticles = (request, response, next) => {
...
-module.exports = { getArticle, getArticles };
\ No newline at end of file
+const patchArticle = (request, response, next) => {
+    const {article_id} = request.params;
+    const {inc_votes} = request.body;
+    selectArticle(article_id).then((article) => {
+        updateArticle(article.article_id, inc_votes).then((updatedArticle) => {
+            response.status(201).send({article: updatedArticle})
+        })
+        .catch((err) => {
```
Comment (Hard-Luck) [inline controllers/articles-controller.js]: You should only need one catch block

File: models/articles-model.js [hunk 4]
```diff
@@ -36,4 +36,12 @@ const selectArticles = () => {
         })
 };
 
-module.exports = { selectArticle, selectArticles }
\ No newline at end of file
+const updateArticle = (article_id, inc_votes) => {
+    return connection.query(`UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *;`, [inc_votes, article_id])
```
Comment (Hard-Luck) [inline models/articles-model.js:40]: very good

File: __tests__/api.test.js (hunk 2 above)
Comment (Hard-Luck) [reply __tests__/api.test.js:341]: and does this work for {inc_votes: -1} ?

**Review by Hard-Luck — CHANGES_REQUESTED** [review]
You just need to add a couple of tests then resubmit

**Review by B7Bprog — APPROVED** [review]
Hey Kristina, nice work, please merge.

**Pull request merged**